nethttp/server.go

```diff
@@ -68,12 +59,11 @@ func Middleware(tr opentracing.Tracer, h http.Handler, options ...MWOption) http
 		ext.HTTPMethod.Set(sp, r.Method)
 		ext.HTTPUrl.Set(sp, r.URL.String())
 		ext.Component.Set(sp, "net/http")
-		w = &statusCodeTracker{w, 200}
 		r = r.WithContext(opentracing.ContextWithSpan(r.Context(), sp))
 
-		h.ServeHTTP(w, r)
+		m := httpsnoop.CaptureMetrics(h, w, r)
```

Comment:
one thing about httpsnoop is that its API forces a lot more allocations than necessary. Using CaptureMetrics when you only need status code may be even more expensive. Could you try a mem benchmark before & after?

Reply:
Would you accept a PR that just makes `statusCodeTracker` implement `http.{CloseNotifier,Flusher,Pusher,Hijacker}` by calling those interfaces' methods on the wrapped `ResponseWriter`? (I.e., without pulling in all of `httpsnoop`.)

(Motivation: At @sourcegraph we need to be able to flush the response body from a handler whose `ResponseWriter` is wrapped in `statusCodeTracker`.)

Reply:
I'm not sure if you think this is a good idea, but I'm partial to this idea I suggested on the go issue about capturing status codes https://github.com/golang/go/issues/18997#issuecomment-361059962 You can just add a method `Wrap() http.ResponseWriter` that returns the underlying `http.ResponseWriter` and it could become a defacto standard. Or maybe we should choose a better name first.